src/GHC/RTS/Events/Binary.hs

```diff
@@ -1136,9 +1137,10 @@ putEventSpec (WakeupThread t c) = do
     putCap c
 
 putEventSpec (ThreadLabel t l) = do
-    putE (fromIntegral (T.length l) + sz_tid :: Word16)
+    let l' = TE.encodeUtf8 l
```

Comment:
Personally I prefer to use ViewPatterns like `ThreadLabel t (TE.encodeUtf8 -> l)` to reduce the number of variables but it's a minor point.